build_modules/test/meta_module_builder_test.dart

```diff
@@ -26,4 +27,62 @@ main() {
           encodedMatchesMetaModule(metaA),
     });
   });
+
+  group('isSupported', () {
+    test('is false for libraries that import dart:io on web', () async {
+      var assetA = AssetId('a', 'lib/a.dart');
+      var assetB = AssetId('a', 'lib/b.dart');
+      var moduleA =
+          Module(assetA, [assetA], <AssetId>[], DartPlatform.dart2js, true);
+      var moduleB =
+          Module(assetB, [assetB], <AssetId>[], DartPlatform.dart2js, false);
+      var metaA = MetaModule([moduleA, moduleB]);
+      await testBuilder(MetaModuleBuilder(DartPlatform.dart2js), {
+        'a|lib/a.module.library':
+            ModuleLibrary.fromSource(assetA, '').serialize(),
+        'a|lib/b.module.library':
+            ModuleLibrary.fromSource(assetB, "import 'dart:io';").serialize(),
+      }, outputs: {
+        'a|lib/${metaModuleExtension(DartPlatform.dart2js)}':
+            encodedMatchesMetaModule(metaA),
+      });
+    });
+
+    test('is false for connected components that import dart:io on web',
+        () async {
+      var assetA = AssetId('a', 'lib/a.dart');
+      var assetB = AssetId('a', 'lib/b.dart');
+      var moduleA = Module(
+          assetA, [assetA, assetB], <AssetId>[], DartPlatform.dart2js, false);
+      var metaA = MetaModule([moduleA]);
+      await testBuilder(MetaModuleBuilder(DartPlatform.dart2js), {
+        'a|lib/a.module.library':
+            ModuleLibrary.fromSource(assetA, "import 'b.dart';").serialize(),
+        'a|lib/b.module.library': ModuleLibrary.fromSource(
+                assetB, "import 'dart:io';import 'a.dart';")
+            .serialize(),
+      }, outputs: {
+        'a|lib/${metaModuleExtension(DartPlatform.dart2js)}':
+            encodedMatchesMetaModule(metaA),
+      });
+    });
+
+    test('is false for coarse modules that import dart:io on web', () async {
+      var assetA = AssetId('a', 'lib/a.dart');
+      var assetB = AssetId('a', 'lib/src/b.dart');
+      var moduleA = Module(
+          assetA, [assetA, assetB], <AssetId>[], DartPlatform.dart2js, false);
+      var metaA = MetaModule([moduleA]);
+      await testBuilder(MetaModuleBuilder(DartPlatform.dart2js), {
+        'a|lib/a.module.library':
+            ModuleLibrary.fromSource(assetA, "import 'src/b.dart';")
+                .serialize(),
+        'a|lib/src/b.module.library':
+            ModuleLibrary.fromSource(assetB, "import 'dart:io';").serialize(),
+      }, outputs: {
+        'a|lib/${metaModuleExtension(DartPlatform.dart2js)}':
+            encodedMatchesMetaModule(metaA),
+      });
+    });
```

Comment:
Maybe add a test to show that `isSupported` doesn't bubbles up transitively between modules?

Reply:
done